src/command.py

```diff
@@ -110,6 +116,26 @@ def execute(self):
             'components': [],
             'dependencies': [],
         }
+
+        required_ids = set()
+        if self._arguments.exclude_dev:
+            to_visit: List[Node] = []
+            visited_ids = set()
+            node: Node
+            for node in deps_graph.nodes:
```

Comment:
can be shortened like 
```py
to_visit = set(node in deps_graph.nodes if node.ref is None)
```
now `to_visit` is a (unique) set, not a list, this saves comparisons later. 
you will be able to traverse it, simply.

Reply:
I changed it as suggested.